ern-core/src/MavenUtils.ts

```diff
@@ -42,7 +42,15 @@ export default class MavenUtils {
     } else if (repoType === 'http') {
       mavenUser = mavenUser ? `'${mavenUser}'` : 'mavenUser'
       mavenPassword = mavenPassword ? `'${mavenPassword}'` : 'mavenPassword'
-      return `repository(url: "${mavenRepositoryUrl}") { authentication(userName: ${mavenUser}, password: ${mavenPassword}) }`
+      let authBlock = ''
+      if (mavenUser && mavenPassword) {
```

Comment:
This if statement will always evaluate to true. The other `else if` blocks are unreachable.
`mavenUser = mavenUser ? ``'${mavenUser}'`` : 'mavenUser'`

If `mavenUser` variable is undefined or null, it will be set to string `mavenUser`
If `mavenUser` variable is an empty string '', it will be set to string `mavenUser`
If `mavenUser` variable is a non empty string, for ex `user`, the variable will be set to string `user`

Naming of these variable leads to confusion (my bad), should have named them `userName` and `password` to be consistent with the name of the properties of the maven plugin. Can always rename, but does not change that the `else if` blocks will still be unreachable. Let's discuss design and implementation. We should also add some unit tests to this one, as the method is quite straightforward to test.